src/main/java/com/powsybl/openloadflow/NetworkCache.java

```diff
@@ -0,0 +1,367 @@
+import com.powsybl.openloadflow.network.VoltageControl;
+import com.powsybl.openloadflow.network.impl.LfNetworkList;
+import com.powsybl.openloadflow.network.util.PreviousValueVoltageInitializer;
+import com.powsybl.openloadflow.util.PerUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.lang.ref.WeakReference;
+import java.util.*;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public enum NetworkCache {
+    INSTANCE;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetworkCache.class);
+
+    public static class Entry extends DefaultNetworkListener {
+
+        private final WeakReference<Network> networkRef;
+
+        private final LoadFlowParameters parameters;
+
+        private List<AcLoadFlowContext> contexts;
+
+        private LfNetworkList.VariantCleaner variantCleaner;
+
+        public Entry(WeakReference<Network> networkRef, LoadFlowParameters parameters) {
+            this.networkRef = Objects.requireNonNull(networkRef);
+            this.parameters = parameters;
+        }
+
+        public WeakReference<Network> getNetworkRef() {
+            return networkRef;
+        }
+
+        public List<AcLoadFlowContext> getContexts() {
+            return contexts;
+        }
+
+        public void setContexts(List<AcLoadFlowContext> contexts) {
+            this.contexts = contexts;
+        }
+
+        public void setVariantCleaner(LfNetworkList.VariantCleaner variantCleaner) {
+            this.variantCleaner = variantCleaner;
+        }
+
+        public LoadFlowParameters getParameters() {
+            return parameters;
+        }
+
+        private void reset() {
+            if (contexts != null) {
+                for (AcLoadFlowContext context : contexts) {
+                    context.close();
+                }
+                contexts = null;
+            }
+        }
+
+        private void onStructureChange() {
+            // too difficult to update LfNetwork incrementally
+            reset();
+        }
+
+        @Override
+        public void onCreation(Identifiable identifiable) {
+            onStructureChange();
+        }
+
+        @Override
+        public void afterRemoval(String s) {
+            onStructureChange();
+        }
+
+        private static Bus getBus(Injection<?> injection, AcLoadFlowContext context) {
+            return context.getParameters().getNetworkParameters().isBreakers()
+                    ? injection.getTerminal().getBusBreakerView().getBus()
+                    : injection.getTerminal().getBusView().getBus();
+        }
+
+        private boolean onLoadUpdate(Load load, String attribute, Object oldValue, Object newValue) {
+            boolean found = false;
+            for (AcLoadFlowContext context : contexts) {
```

Comment:
Finally, I think that for a first version the load support is not needed. It complexes the feature and it is not a real need for the IA use cases I know.

Reply:
Ok, discarded from the PR